root/edit/details/EditArtist.js

```diff
@@ -0,0 +1,183 @@
+/*
+ * @flow
+ * Copyright (C) 2020 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {
+  artistBeginAreaLabel,
+  artistBeginLabel,
+  artistEndAreaLabel,
+  artistEndLabel,
+} from '../../artist/utils';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import FullChangeDiff from '../../static/scripts/edit/components/edit/FullChangeDiff';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+import {commaOnlyListText} from '../../static/scripts/common/i18n/commaOnlyList';
+
+type EditArtistProps = {
+  +display_data: {
+    +area?: CompT<AreaT | null>,
+    +artist: ArtistT,
+    +begin_area?: CompT<AreaT | null>,
+    +begin_date?: CompT<PartialDateT | null>,
+    +comment?: CompT<string>,
+    +end_area?: CompT<AreaT | null>,
+    +end_date?: CompT<PartialDateT | null>,
+    +ended?: CompT<boolean>,
```

Comment:
You'll need to `boolean_to_json` the old/new values if it exists, otherwise you can get something like `{ new: '1', old: '0' }` which is truthy on both sides. I'm sure you've done something similar in your recent PRs. :)